p2p/switch.go

```diff
@@ -410,12 +415,8 @@ func (sw *Switch) stopAndRemovePeer(peer Peer, reason interface{}) {
 		sw.Logger.Error("error while stopping peer", "error", err) // TODO: should return error to be handled accordingly
 	}
 	schema.WritePeerUpdate(sw.traceClient, string(peer.ID()), schema.PeerDisconnect, fmt.Sprintf("%v", reason))
-	for _, reactor := range sw.reactors {
-		reactor.RemovePeer(peer, reason)
-		chs := reactor.GetChannels()
-		if len(chs) > 0 {
-			_ = sw.peerSetByChID[chs[0].ID].Remove(peer) // TODO(tzdybal): error handling
-		}
+	if reason != nil {
```

Comment:
is this a contract we want to enforce: if the reason != nil, remove from all reactors? if so, we should document it in the `stopAndRemovePeer` IMO

Reply:
Yes, it's safe to assume that errors reported by reactors are caused by serious issues or malicious behavior, and it's safer just to drop connection with peer completely.